lib/core/src/Cardano/Wallet/Primitive/Model.hs

```diff
@@ -556,4 +567,11 @@ changeUTxO
     -> UTxO
 changeUTxO pending = evalState $
     mconcat
-    <$> mapM (filterByAddressM isOurAddress . utxoFromTx) (Set.toList pending)
+    <$> mapM (filterByAddressM isOurAddress . mkUTxOFromTx) (Set.toList pending)
+    where
+        -- Generate a UTxO from an transaction, assuming that it passes phase-2
+        -- script validation. Crucially, utxoFromTx will exclude failed
+        -- transactions, hence we define our own function.
```

Comment:
This leaves me with a question:

For transactions in the pending set, what meaning can we ascribe to the `txScriptValidity` field? It seems that at this stage, we wouldn't have enough information to assign a meaningful value. (Unless I'm mistaken?)

The fact that we must ignore the `txScriptValidity` field in the `changeUTxO` function indicates to me that there might be a better way to represent the concept of "not-yet-known" or "unknown".

For example, we could give the `TxScriptValidity` type just two options (`TxScriptValid` and `TxScriptInvalid`), and then give `txScriptValidity` a type of `Maybe TxScriptValidity`. The `Nothing` option would represent the case where we don't yet know. But this also seems non-ideal.